VisualStudioPlugin/SolutionExplorerMenuCommand.cs

```diff
@@ -163,47 +156,121 @@ private void SaveToQuantConnectCallback(object sender, EventArgs e)
+            var filesUploaded = 0;
+            var filesNotUploaded = 0;
             foreach (var file in files)
             {
                 api.DeleteProjectFile(selectedProjectId, file.FileName);
                 var fileContent = File.ReadAllText(file.FilePath);
-                api.AddProjectFile(selectedProjectId, file.FileName, fileContent);
+                var response = api.AddProjectFile(selectedProjectId, file.FileName, fileContent);
+                if (response.Success)
+                {
+                    filesUploaded++;
+                }
+                else
+                {
+                    VSActivityLog.Error("Failed to add project file " + file.FileName);
+                    filesNotUploaded++;
+                }
             }
+            // Update Status bar accordingly based on counters
+            var message = "Files upload complete";
+            message += (filesUploaded != 0) ? ". Uploaded " + filesUploaded : "";
+            message += (filesNotUploaded != 0) ? ". Failed to upload " + filesNotUploaded : "";
+            VsUtils.DisplayInStatusBar(_serviceProvider, message);
+
+            // Return false if any file failed to be uploaded
+            var res = filesNotUploaded == 0;
+            if (!res)
+            {
+                VsUtils.ShowMessageBox(_serviceProvider, "Upload Files Failed", message);
+            }
+            return res;
         }
 
-        private Api.Compile CompileProjectOnServer(int projectId)
+        /// <summary>
+        /// Compiles specific projectId at QuantConnect
+        /// </summary>
+        /// <param name="projectId">Target project Id</param>
+        /// <returns>Returns empty if failed else compile Id</returns>
+        private string CompileProjectOnServer(int projectId)
         {
+            VsUtils.DisplayInStatusBar(_serviceProvider, "Compiling project ...");
             var api = AuthorizationManager.GetInstance().GetApi();
             var compileStatus = api.CreateCompile(projectId);
             var compileId = compileStatus.CompileId;
             while (compileStatus.State == Api.CompileState.InQueue)
             {
-                Thread.Sleep(5000);
+                Thread.Sleep(2500);
                 compileStatus = api.ReadCompile(projectId, compileId);
             }
-            return compileStatus;
+
+            if (compileStatus.State == Api.CompileState.BuildError)
+            {
+                // Default to show Errors, now it is coming empty so use Logs. Will only show First Error || Log
+                var error = compileStatus.Errors.Count == 0 ?
+                    compileStatus.Logs.FirstOrDefault() : compileStatus.Errors.FirstOrDefault();
+                var message = "Compilation error \"" + error + "\"";
+                VsUtils.DisplayInStatusBar(_serviceProvider, message);
+                VsUtils.ShowMessageBox(_serviceProvider, "Compilation Error", error);
```

Comment:
There's often a nice stack trace here. It is probably worthwhile making a nicer error dialog which displays the error message in Courier-New font as a stack trace in a text area.

Reply:
Done, added new Error dialog with compilation error message